build/jenkins_e2e.sh

```diff
@@ -61,7 +61,15 @@ golden_nodes=(
 
 max_retries=8
 
+if [[ -a "${CADVISOR_SERVICE_ACCOUNT:-}" ]] ; then
+  gcloud auth activate-service-account --key-file="${CADVISOR_SERVICE_ACCOUNT}"
+fi
+
 ./runner --logtostderr --test-retry-count=$max_retries \
   --test-retry-whitelist=integration/runner/retrywhitelist.txt \
-  --ssh-options "-i /var/lib/jenkins/gce_keys/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no" \
+  --ssh-options "--project=cadvisor-e2e --strict-host-key-checking=false" \
   ${golden_nodes[*]}
+
```

Comment:
nit: add a comment say we need to switch back using this service account for uploading logs etc